salesforce/connector.go

```diff
@@ -50,27 +46,42 @@ func NewConnector(opts ...Option) (conn *Connector, outErr error) {
 		return nil, err
 	}
 
-	restApi, ok := providerInfo.GetOption(providerOptionRestApiURL)
-	if !ok {
-		return nil, fmt.Errorf("restApiUrl not set: %w", providers.ErrProviderOptionNotFound)
-	}
-
-	domain, ok := providerInfo.GetOption(providerOptionDomain)
-	if !ok {
-		return nil, fmt.Errorf("domain not set: %w", providers.ErrProviderOptionNotFound)
-	}
-
 	conn = &Connector{
-		BaseURL: restApi,
-		Domain:  domain,
 		Client: &common.JSONHTTPClient{
 			HTTPClient: params.Client.Caller,
 		},
 	}
 
-	conn.Client.HTTPClient.Base = providerInfo.BaseURL
+	conn.setBaseURL(providerInfo.BaseURL)
 	conn.Client.HTTPClient.ErrorHandler = conn.interpretError
 	conn.Client.ErrorPostProcessor.Process = handleError
 
 	return conn, nil
 }
+
+// Provider returns the connector provider.
+func (c *Connector) Provider() providers.Provider {
+	return providers.Salesforce
+}
+
+// String returns a string representation of the connector, which is useful for logging / debugging.
+func (c *Connector) String() string {
+	return fmt.Sprintf("%s.Connector", c.Provider())
+}
+
+func (c *Connector) getURL(paths ...string) (*urlbuilder.URL, error) {
```

Comment:
I'd rename this to be more explicit about the fact that it is joining against the REST API url, because there are other kinds of suffixes being used in this connector too